src/main/scala/org/broadinstitute/dsde/workbench/sam/google/GoogleExtensions.scala

```diff
@@ -265,6 +265,45 @@ class GoogleExtensions(val directoryDAO: DirectoryDAO, val accessPolicyDAO: Acce
     }
   }
 
+  def getArbitraryPetServiceAccountKey(user: WorkbenchUser): Future[String] = {
+    for {
+      pets <- directoryDAO.getAllPetServiceAccountsForUser(user.id)
+      key <- if(pets.nonEmpty) getPetServiceAccountKey(user, pets.head.id.project) else getDefaultServiceAccountForShellProject(user)
+    } yield key
+  }
+
+  def getArbitraryPetServiceAccountToken(user: WorkbenchUser, scopes: Set[String]): Future[String] = {
+    getArbitraryPetServiceAccountKey(user).flatMap { key =>
+      getAccessTokenUsingJson(key, scopes)
+    }
+  }
+
+  private def getDefaultServiceAccountForShellProject(user: WorkbenchUser): Future[String] = {
+    val projectName = s"shell-${user.id.value}"
```

Comment:
I think we should have something firecloud specific... was thinking whatever our email domain is... how many characters do we have to play with?

Reply:
subject ID is 21 characters so we have 9 characters to play with

could do something like `fc-$env-111111111111111111111`

`env` would have to be truncated to first 5 chars because staging would be too long in that case

Reply:
Is this the first time we're using "shell" projects?  The idea being that we are just creating the shell so that we have some random project in which to create an SA in for the user?

Reply:
yes that's correct

Reply:
👍